drivers/clock_control/Kconfig.nrf

```diff
@@ -23,7 +23,8 @@ if CLOCK_CONTROL_NRF
 
 choice CLOCK_CONTROL_NRF_SOURCE
 	prompt "32KHz clock source"
-	default CLOCK_CONTROL_NRF_K32SRC_XTAL
+	default CLOCK_CONTROL_NRF_K32SRC_XTAL if !ZTEST
+	default CLOCK_CONTROL_NRF_K32SRC_SYNTH if ZTEST
```

Comment:
@pabigot @andyross @nordic-krch 

Since the assumption that `k_busy_wait()` and `k_sleep()` are driven by the same clock is just not valid for Nordic platfroms (and likely for others as well, present or future), would it be better to introduce a new `FREQUENCY_TOLERANCE` variable that the test uses to extend its tolerance and that can be set to `0` for platforms that use the same clock.
Then you can have something like: `check_timeout_drift()` that uses `FREQUENCY_TOLERANCE` in all tests that combine busy waits with sleep.

Reply:
@andyross could you please weigh in here?

Reply:
Comments in https://github.com/zephyrproject-rtos/zephyr/issues/24784#issuecomment-628044358 -- I'd be personally happy with this patch, or its converse.

To nitpick: I guess stylewise I'd prefer a choice of name that conveys the implication and not just the details.  The important thing to an app isn't what whether the source is "SYNTH" or "XTAL", it's that the "synth" choice is subject to temperature and semiconductor process skew where the crystal is tuned to more conventional tolerances.

As far as adding code to the test framework, that would work, but it doesn't seem much different than just adding a fudge factor for individual cases.  These kind of long busy waits are pretty rare even in the tests.